src/redisearch_rs/inverted_index/src/index_result.rs

```diff
@@ -114,6 +120,34 @@ impl RSOffsetVector<'_> {
         self.data = std::ptr::null_mut();
         self.len = 0;
     }
+
+    /// Create a deep copy of this offset vector, allocating new memory for the data. This data
+    /// should be freed using [`Self::free_data()`].
+    pub fn deep_copy(&self) -> RSOffsetVector<'static> {
```

Comment:
Why not just use `Clone`?

Reply:
Would `ToOwned` make more sense then?

Reply:
Made it `into_owned` in https://github.com/RediSearch/RediSearch/commit/0fd3c0658b9cf835cc77b0ba553662126039e577